tools/apps/src/bin/quiche-client.rs

```diff
@@ -0,0 +1,790 @@
+                            f.write_all(&buf[..read]).ok();
+                        }
+
+                        // The server reported that it has no more data to send on
+                        // a client-initiated
+                        // bidirectional stream, which means
+                        // we got the full response. If all responses are received
+                        // then close the connection.
+                        if &s % 4 == 0 && fin {
+                            *reqs_complete += 1;
+                            let reqs_count = reqs.len();
+
+                            debug!(
+                                "{}/{} responses received",
+                                reqs_complete, reqs_count
+                            );
+
+                            if *reqs_complete == reqs_count {
+                                info!(
+                                    "{}/{} response(s) received in {:?}, closing...",
+                                    reqs_complete,
+                                    reqs_count,
+                                    req_start.elapsed()
+                                );
+
+                                match conn.close(true, 0x00, b"kthxbye") {
+                                    // Already closed.
+                                    Ok(_) | Err(quiche::Error::Done) => (),
+
+                                    Err(e) =>
+                                        panic!("error closing conn: {:?}", e),
+                                }
+
+                                break;
+                            }
+                        }
+                    }
+                }
+            },
+
+            HttpConn::Http3Conn {
+                h3_conn,
+                fds,
+                reqs_complete,
+                reqs,
+                ..
+            } => {
+                loop {
+                    match h3_conn.poll(conn) {
+                        Ok((
+                            stream_id,
+                            quiche::h3::Event::Headers { list, .. },
+                        )) => {
+                            info!(
+                                "got response headers {:?} on stream id {}",
+                                list, stream_id
+                            );
+                        },
+
+                        Ok((stream_id, quiche::h3::Event::Data)) => {
+                            if let Ok(read) =
+                                h3_conn.recv_body(conn, stream_id, buf)
+                            {
+                                debug!(
+                                    "got {} bytes of response data on stream {}",
+                                    read, stream_id
+                                );
+
+                                print!("{}", unsafe {
+                                    std::str::from_utf8_unchecked(&buf[..read])
+                                });
```

Comment:
(same as L599) Since we have `--dump-responses` option, I think when we set this option, no need to print the body. to stdout